### ISSUE TITLE
Black border when resizing on iPhone Safari

### ISSUE THREAD

**Issue by edwh** [kartik-v/bootstrap-fileinput#1337]
## Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [ ] The issue still exists against the latest `master` branch of bootstrap-fileinput. (I'm on 4.5.1 which is very recent)
- [X] This is not an usage question. I confirm having read the plugin [documentation](http://plugins.krajee.com/file-input) and [demos](http://plugins.krajee.com/file-input/demo).
- [X] This is not a general programming / coding question. (Those should be directed to the [webtips Q & A forum](http://webtips.krajee.com/questions)).
- [X] I have attempted to find the simplest possible steps to reproduce the issue.
- [ ] I have included a failing test as a pull request (Optional).

## Steps to reproduce the issue

1. Use iPhone Safari
2. Use resizeImage: true, maxImageWidth: 800.  

## Expected behavior and actual behavior

Get a resized photo which looks ok.

When I follow those steps, I see a black border on the image:

![image](https://user-images.githubusercontent.com/56696/49537615-57c98000-f8c1-11e8-870a-847a26cc4e13.png)

I've got the image here by dumping the image posted to the server.

I was expecting no black border, i.e. appropriate image dimensions.  

## Environment

Browsers

- [ ] Google Chrome
- [ ] Mozilla Firefox
- [ ] Internet Explorer
- [X] Safari

Operating System

- [ ] Windows
- [ ] Mac OS X
- [ ] Linux
- [X] Mobile

Libraries

- jQuery version: 3.3.1
- bootstrap-fileinput version: 4.5.1

## Isolating the problem

- [ ] This bug happens [on the plugin demos page](http://plugins.krajee.com/file-input/demo)
- [ ] The bug happens consistently across all tested browsers
- [ ] This bug happens when using bootstrap-fileinput without other plugins
- [ ] I can reproduce this bug in [a jsbin](https://jsbin.com/)

I can only repro this on an iPhone using Safari.

**Comment by kartik-v** [comment]
This is due to an iOS restriction / bug / behavior where it subsamples images over 2Megapixels. According to [this link](https://developer.apple.com/library/content/documentation/AppleApplications/Reference/SafariWebContent/CreatingContentforSafarioniPhone/CreatingContentforSafarioniPhone.html#//apple_ref/doc/uid/TP40006482-SW15), JPEG files over 2M pixels will be subsampled.

Will check later on this when I get time.

**Comment by edwh** [comment]
Thanks.  It would be useful if there was any way to know on the client 
that this might occur, because then I could avoid resizing.  That's not 
ideal, because obvious resizing is good for bandwidth, but it would 
avoid having images that look bad.

------ Original Message ------

**Comment by fstrahberger** [comment]
Hello Edward,
here is a solution that helped me solve this problem:
- use https://github.com/stomita/ios-imagefile-megapixel to resize the image
- disable resize in fileinput config

```
$("#upload_files").fileinput({
        language: 'de',
        previewFileType: "image",
        maxFileCount: 20,
        autoOrientImage: true,
        // maxImageHeight: 1200,
        // maxImageWidth: 1200,
        // resizePreference: 'width',
        // resizeImage: true,
        browseClass: "btn btn-success",
```
- resize the images in fileimagesloaded event (get files from stack, clear stack , refill stack with modified images)

```
$('#upload_files').on('fileimagesloaded ', function(event) {
    var myFiles = $('#upload_files').fileinput('getFileStack');
    $('#upload_files').fileinput('clearStack');

    var imgList = [];
    for (i = 0; i < myFiles.length; i++) {
       let item = {
            mpimg: new MegaPixImage(myFiles[i]),
            file: myFiles[i]
        };
        imgList.push(item);
    }

    imgList.forEach(function(item) {
         var c = document.createElement("canvas");
         item.mpimg.render(c, { maxWidth: 1200, maxHeight: 1200 }, function(i){
             c.toBlob(function(blob){
                 blob.lastModifiedDate = item.file.lastModifiedDate;
                 blob.name = item.file.name;
                 blob.type = item.file.type;
                 $('#upload_files').fileinput('addToStack', blob);
             }, item.file.type, 0.75);
         });
    });
});
```

**Comment by silentcossack** [comment]
I can confirm this bug.  
This only happens when taking a photo on iPhone in portrait mode. Landscape photos are fine. I believe the code does not properly detect image orientation and resizes improper dimensions.

**Comment by kartik-v** [comment]
The orientation validation specific to iOS safari browser would be resolved by #1362.
